addons/mass_mailing/tests/test_mailing_ab_testing.py

```diff
@@ -103,11 +103,23 @@ def test_mailing_ab_testing_campaign(self):
         })
         ab_mailing.invalidate_recordset()
 
-        # Check if the campaign is correclty created and the values set on the mailing are still the same
+        # Check if the campaign is correctly created and the values set on the mailing are still the same
         self.assertTrue(ab_mailing.campaign_id, "A campaign id is present for the A/B test mailing")
         self.assertEqual(ab_mailing.ab_testing_winner_selection, 'manual', "The selection winner has been propagated correctly")
         self.assertEqual(ab_mailing.ab_testing_schedule_datetime, schedule_datetime, "The schedule date has been propagated correctly")
 
+        # Check that while enabling the A/B testing, if campaign is already set, new one should not be created
+        created_mailing_campaign = ab_mailing.campaign_id
+        ab_mailing.ab_testing_enabled = False
+        ab_mailing.ab_testing_enabled = True
+        self.assertEqual(ab_mailing.campaign_id, created_mailing_campaign, "No new campaign should have been created")
+
+        # Check that while enabling the A/B testing, if user manually selects a campaign, it should be saved
+        # rather than being replaced with the automatically created one
+        ab_mailing.write({'ab_testing_enabled': False, 'campaign_id': False})
+        ab_mailing.write({'ab_testing_enabled': True, 'campaign_id': created_mailing_campaign})
+        self.assertEqual(ab_mailing.campaign_id, created_mailing_campaign, "No new campaign should have been created")
```

Comment:
Looks like a check is missing to make sure we create a campaign if it's missing when "writing" (currently the test only checks it creates the missing campaign for the "create" operation).